Makefile

```diff
@@ -72,7 +72,10 @@ TOOLSPATH=$(BUILDPATH)/tools
 CORE_PATH=$(BUILDPATH)/src/core
 PORTAL_PATH=$(BUILDPATH)/src/portal
 CHECKENVCMD=checkenv.sh
-
+ARCH ?= $(shell uname -m)
```

Comment:
From a build and engineering perspective, we should define the ARCH in a single place and pass it to all components that need it, rather than retrieving it separately in multiple locations.